docs/essentials/gc/unmanaged/usingobjects.md

```diff
@@ -1 +1,153 @@
+
+The following example uses the `using` statement to create and release a [System.IO.StreamReader](http://dotnet.github.io/api/System.IO.StreamReader.html) object.
+
+```csharp
+using System;
+using System.IO;
+
+public class Example
+{
+   public static void Main()
+   {
+      Char[] buffer = new Char[50];
+      using (StreamReader s = new StreamReader("File1.txt")) {
+         int charsRead = 0;
+         while (s.Peek() != -1) {
+            charsRead = s.Read(buffer, 0, buffer.Length);
+            //
+            // Process characters read.
+            //   
+         }
+         s.Close();    
+      }
+
+   }
+}
+```
+
+Note that although the `StreamReader` class implements the `IDisposable` interface, which indicates that it uses an unmanaged resource, the example doesn't explicitly call the `StreamReader.Dispose` method. When the compiler encounters the `using` statement, it emits intermediate language (IL) that is equivalent to the following code that explicitly contains a `try/finally` block. 
+
+```csharp
+using System;
+using System.IO;
+
+public class Example
+{
+   public static void Main()
+   {
+      Char[] buffer = new Char[50];
+      {
+         StreamReader s = new StreamReader("File1.txt"); 
+         try {
+            int charsRead = 0;
+            while (s.Peek() != -1) {
+               charsRead = s.Read(buffer, 0, buffer.Length);
+               //
+               // Process characters read.
+               //   
+            }
+            s.Close();
+         }
+         finally {
+            if (s != null)
+               ((IDisposable)s).Dispose();     
+         }       
+      }
+   }
+}
+```
+
+The `using` statement also allows you to acquire multiple resources in a single statement, which is internally equivalent to nested `using` statements. The following example instantiates two `StreamReader` objects to read the contents of two different files. 
+
+```csharp
+using System;
+using System.IO;
+
+public class Example
+{
+   public static void Main()
+   {
+      Char[] buffer1 = new Char[50], buffer2 = new Char[50];
+
+      using (StreamReader version1 = new StreamReader("file1.txt"),
+                          version2 = new StreamReader("file2.txt")) {
+         int charsRead1, charsRead2 = 0;
+         while (version1.Peek() != -1 && version2.Peek() != -1) {
+            charsRead1 = version1.Read(buffer1, 0, buffer1.Length);
+            charsRead2 = version2.Read(buffer2, 0, buffer2.Length);
+            //
+            // Process characters read.
+            //
+         }
+         version1.Close();
+         version2.Close();
+      }
+   }
+}
+```
+
+## Try/finally block
+
+Instead of wrapping a `try/finally` block in a `using` statement, you may choose to implement the `try/finally` block directly. This may be your personal coding style, or you might want to do this for one of the following reasons: 
```

Comment:
I don't think "wrapping" is the right term to use here. Not sure what would be better though.